packages/studio-base/src/panels/CallService/CallService.tsx

```diff
@@ -0,0 +1,247 @@
+import { Config } from "@foxglove/studio-base/panels/CallService/types";
+import { fonts } from "@foxglove/studio-base/util/sharedStyleConstants";
+
+import { defaultConfig, settingsActionReducer, useSettingsTree } from "./settings";
+
+const log = Log.getLogger(__dirname);
+
+type Props = {
+  context: PanelExtensionContext;
+};
+
+type State = {
+  status: "requesting" | "error" | "success";
+  value: string;
+};
+
+const useStyles = makeStyles<{ buttonColor?: string }>()((theme, { buttonColor }) => {
+  const augmentedButtonColor = buttonColor
+    ? theme.palette.augmentColor({
+        color: { main: buttonColor },
+      })
+    : undefined;
+
+  return {
+    button: {
+      backgroundColor: augmentedButtonColor?.main,
+      color: augmentedButtonColor?.contrastText,
+
+      "&:hover": {
+        backgroundColor: augmentedButtonColor?.dark,
+      },
+    },
+    textarea: {
+      height: "100%",
+
+      [`.${inputBaseClasses.root}`]: {
+        width: "100%",
+        height: "100%",
+        textAlign: "left",
+        backgroundColor: theme.palette.background.paper,
+        overflow: "hidden",
+        padding: theme.spacing(1, 0.5),
+
+        [`.${inputBaseClasses.input}`]: {
+          height: "100% !important",
+          lineHeight: 1.4,
+          fontFamily: fonts.MONOSPACE,
+          overflow: "auto !important",
+          resize: "none",
+        },
+      },
+    },
+  };
+});
+
+function parseInput(value: string): { error?: string; parsedObject?: unknown } {
+  let parsedObject;
+  let error = undefined;
+  try {
+    const parsedAny: unknown = JSON.parse(value);
+    if (Array.isArray(parsedAny)) {
+      error = "Request content must be an object, not an array";
+    } else if (parsedAny == null /* eslint-disable-line no-restricted-syntax */) {
+      error = "Request content must be an object, not null";
+    } else if (typeof parsedAny !== "object") {
+      error = `Request content must be an object, not ‘${typeof parsedAny}’`;
+    } else {
+      parsedObject = parsedAny;
+    }
+  } catch (e) {
+    error = value.length !== 0 ? e.message : "Enter valid request content as JSON";
+  }
+  return { error, parsedObject };
+}
+
+export function CallService({ context }: Props): JSX.Element {
+  // panel extensions must notify when they've completed rendering
+  // onRender will setRenderDone to a done callback which we can invoke after we've rendered
+  const [renderDone, setRenderDone] = useState<() => void>(() => () => {});
+  const [state, setState] = useState<State | undefined>();
+  const [config, setConfig] = useState<Config>(() => ({
+    ...defaultConfig,
+    ...(context.initialState as Partial<Config>),
+  }));
+  const { classes } = useStyles({ buttonColor: config.buttonColor });
+
+  useEffect(() => {
+    context.saveState(config);
+    context.setDefaultPanelTitle(
+      config.serviceName === "" ? undefined : `Call service ${config.serviceName}`,
```

Comment:
Want to avoid "Call service undefined":
<img width="167" alt="image" src="https://github.com/foxglove/studio/assets/14237/7252a85c-b9a3-4c24-9c28-24238e764153">

```suggestion
      config.serviceName ? `Call service ${config.serviceName}` : undefined,
```